db/migrate/20210705035105_add_results_to_keywords.rb

```diff
@@ -0,0 +1,27 @@
+class AddResultsToKeywords < ActiveRecord::Migration[6.1]
+  def change
+    add_column :keywords, :status, :integer, default: 0, null: false
+    add_column :keywords, :ads_top_count, :integer
+    add_column :keywords, :ads_page_count, :integer
+    add_column :keywords, :non_ads_result_count, :integer
+    add_column :keywords, :total_link_count, :integer
+    add_column :keywords, :html, :text
+
+    add_index :keywords, :status
+    add_index :keywords, :ads_top_count
+    add_index :keywords, :ads_page_count
+    add_index :keywords, :non_ads_result_count
+    add_index :keywords, :total_link_count
+
+    create_table :result_links do |t|
```

Comment:
Creating seperate table like this is also fine, but maybe you can also try postgres array type or postgres jsonb type to store results in keywords table itself.
```
    add_column :keywords, :result_links, :jsonb

    result_links: {
    "non_ads": ["", ""],
    "all_ads": ["", ""],
     "top_ads": ["", ""],
}

Reply:
Thanks for this feedback @ankitkalia1195 ! :) 
For JSON/JSONB, I have researched before hands and it is not recommended:
- several articles points out the limitations of DB side calculations possible, which made me think of the advance query needed in coming user stories
- [Nimble compass](https://nimblehq.co/compass/development/code-conventions/ruby-on-rails/#relational-database) recommend `jsonb` for configurations like data which seems far different as link data will be needed by the search feature

Regarding Array type, I need to research more :) 
I'll come back with what I found. 

Though let me know if you disagree with the jsonb, you might have real experiences that prove the contrary and I'd be happy to hear from them (I've only searched for 15 minutes ~ )

Reply:
Sure, jsonb can make the search more complex if you want to search by urls. :). So we can keep seperate table.

Reply:
Ok, I will keep separate tables, but I will spend some times to practice with jsonb and arrays, for learning purpose (in a local branch only). 
Thanks a lot!! :)